### ISSUE TITLE
ForwardingGestureRecognizer blocked by other gesture recognizers on iOS

### ISSUE THREAD

**Issue by cyanglaz** [flutter/flutter#61991]
If there is a gesture recognizer on the platform view, which implements `- (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer
    shouldBeRequiredToFailByGestureRecognizer:(UIGestureRecognizer*)otherGestureRecognizer` and returns YES, the ForwardingGestureRecognizer will be blocked, which broke the pointer event sequence that sent to the `FlutterViewController`.

One of the issues that is related to this failure is with WKWebView

>The WKDeferringGestureRecognizer(Webkit Internal API) in the WKWebView seems to be the one that is blocking the ForwardingGestureRecognizer in the flutter engine.
>https://github.com/WebKit/webkit/blame/master/Source/WebKit/UIProcess/ios/WKDeferringGestureRecognizer.mm
>This is a new gesture recognizer that was added about 7 months ago.

related to https://github.com/flutter/flutter/issues/53490

**Comment by blasten** [comment]
cc @pcsosinski

**Comment by pcsosinski** [comment]
is this a regression from 1.17?

**Comment by blasten** [comment]
My understanding is that this bug has been there prior to 1.17, but @cyanglaz to confirm. One of our iOS developers brought this issue a couple times.

**Comment by cyanglaz** [comment]
@pcsosinski No. There has always been a possibility an iOS platform view blocks the forwarding gesture recognizer. We just haven't encountered an actual case until WKWebView introduced `WkDeferringGestureRecognizer` in iOS 13.4.
And it also affects other platform views if similar logics to `WkDeferringGestureRecognizer` are implemented.

**Comment by cyanglaz** [comment]
Discovered that the root cause of the p1 issue is https://github.com/flutter/flutter/issues/46167, not this. Lowering the priority of this to p3 and will work on the other one first.

**Comment by ztigunes** [comment]
Other webview packages  also not working on ios. We had to delay the ios deployment because of this.  
webview_flutter not updated since May 21.  Do you have a road map for this bug? Thanks. Best regards. (flutter stable channel  1.20.2)

**Comment by seeuli** [comment]
I encountered a new crash, the following is the stack information. @jason-simmons  @jmagman  @gspencergoog 

```
Thread 0 Crashed:
0   libobjc.A.dylib                 objc_msgSend + 16
1   Flutter                         -[ForwardingGestureRecognizer touchesBegan:withEvent:] (FlutterPlatformViews.mm:642)
2   UIKitCore                       -[UIGestureRecognizer _touchesBegan:withEvent:] + 252
3   UIKitCore                       -[UITouchesEvent _sendEventToGestureRecognizer:] + 288
4   UIKitCore                       __47-[UIGestureEnvironment _updateForEvent:window:]_block_invoke + 68
5   UIKitCore                       -[UIGestureEnvironment _deliverEvent:toGestureRecognizers:usingBlock:] + 220
6   UIKitCore                       -[UIGestureEnvironment _updateForEvent:window:] + 204
7   UIKitCore                       -[UIWindow sendEvent:] + 3112
8   UIKitCore                       -[UIApplication sendEvent:] + 340
9   UIKitCore                       __dispatchPreprocessedEventFromEventQueue + 1768
10  UIKitCore                       __handleEventQueueInternal + 4828
11  UIKitCore                       __handleHIDEventFetcherDrain + 152
12  CoreFoundation                  __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
13  CoreFoundation                  __CFRunLoopDoSource0 + 88
14  CoreFoundation                  __CFRunLoopDoSources0 + 176
15  CoreFoundation                  __CFRunLoopRun + 1004
16  CoreFoundation                  CFRunLoopRunSpecific + 436
17  GraphicsServices                GSEventRunModal + 104
18  UIKitCore                       UIApplicationMain + 212
```

flutter doctor
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.9.1+hotfix.107, on Mac OS X 10.15.6 19G2021, locale zh-Hans-CN)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
   ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses
[!] Xcode - develop for iOS and macOS (Xcode 12.0.1)
   ! CocoaPods out of date (1.6.0 is recommended).
       CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
       Without CocoaPods, plugins will not work on iOS or macOS.
       For more info, see https://flutter.dev/platform-plugins
     To upgrade:
       sudo gem install cocoapods
[✓] Android Studio (version 3.6)
[!] Connected device
   ! No devices available

! Doctor found issues in 3 categories.
```

**Comment by cyanglaz** [comment]
@seeuli Thank you for reporting this! This looks like a new issue. I opened a new issue tracking this crash: https://github.com/flutter/flutter/issues/67933

**Comment by seeuli** [comment]
> @seeuli Thank you for reporting this! This looks like a new issue. I opened a new issue tracking this crash: #67933

Thank you!

**Comment by wanjm** [comment]
why is there no one take care of as it has been p3 for one year;

**Comment by JsouLiang** [comment]
> 
`However, if the platform view contains custom gesture recognizers, which overrides `touchesBegan:` and does something with the gesture, this is going to be problematic.` do you know the deep reason? 

`To decide if `DelayingGestureRecognizer` should fail, we send the touch event to the framework through `ForwardingGestureRecognizer` (PlatformThread). When the gesture arena makes a decision (block platform view or release platform view), it sends the result back through the method channel asynchronously.` And may be we can use the `ffi channel` to solve the problem? https://docs.google.com/document/d/1QhCs6RnDHG9ltYDOx9jaPg-s4sxOTFMC_kt3AZdf25g/edit#heading=h.k87q5jlot4cj

**Comment by JsouLiang** [comment]
@cyanglaz About this issue, do we have a plan to fix it?
